crypto/CMakeLists.txt

```diff
@@ -6,6 +6,10 @@ if (MSVC)
         set(OPENSSL_SYS_DEFINE "OPENSSL_SYS_WIN32")
     elseif (OPENSSL_TARGET_ARCH STREQUAL "x64")
         set(OPENSSL_SYS_DEFINE "OPENSSL_SYS_WIN64A")
+    elseif (OPENSSL_TARGET_ARCH STREQUAL "arm64")
+        set(OPENSSL_SYS_DEFINE "OPENSSL_SYS_WIN64ARM64")
+    elseif (OPENSSL_TARGET_ARCH STREQUAL "arm64ec")
```

Comment:
I passed in : OPENSSL_TARGET_ARCH to -D

Reply:
Please don't. It will disable ASM by the time OpenSSL supports it.
Consider also on macOS / iOS when `-DCMAKE_OSX_ARCHITECTURES=arm64e` is specified this value is calculated to be `arm64`.

Reply:
Also at around line 57

```
#define RC4_INT unsigned @OPENSSL_RC4_INT@
```
```
#cmakedefine BN_LLONG
```

This should match the definition in `openssl/Configurations/50-win-onecore.conf`, or built binary is not compatible with the one built by original perl-based Configure